### PR TITLE
Add support for preview image in media serializer

### PR DESCRIPTION
**Pull request by alexandersch**
I've added support for the preview image in MediaSerializer as it didn't respect preview images.

### CODE REVIEW

File: Content/Serializer/MediaSerializer.php [hunk 1]
```diff
@@ -80,21 +80,26 @@ public function serialize(MediaInterface $media, string $locale, ?SerializationC
         unset($mediaData['adminUrl']);
         unset($mediaData['_hash']);
 
+        $formatMediaApi = $apiMedia;
+        if($media->getPreviewImage()) {
+            $formatMediaApi = new Media($media->getPreviewImage(), $locale);
```
Comment (alexander-schranz) [inline Content/Serializer/MediaSerializer.php]: Looks like there is currently a false typehint in the sulu core here so we can workaround this currently with:

```suggestion
        /** @var Media|null $previewImage */
        $previewImage = $media->getPreviewImage();
        if($previewImage) {
            $formatMediaApi = new Media($previewImage, $locale);
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
@alexandersch Thank you for the Pull Request 👍 . Would be nice to have a Unit test with a `PreviewImage` in https://github.com/sulu/SuluHeadlessBundle/blob/0.x/Tests/Unit/Content/Serializer/MediaSerializerTest.php. Let me know if you have time to add this else we can do it.

File: Content/Serializer/MediaSerializer.php [hunk 2]
```diff
@@ -80,21 +80,28 @@ public function serialize(MediaInterface $media, string $locale, ?SerializationC
         unset($mediaData['adminUrl']);
         unset($mediaData['_hash']);
 
+        $formatMediaApi = $apiMedia;
+        /** @var Media|null $previewImage */
```
Comment (alexander-schranz) [inline Content/Serializer/MediaSerializer.php]: Sorry my fault this need to be as it is not an api media instead the media entity:

```suggestion
        /** @var MediaInterface|null $previewImage */
```

**Comment by alexandersch** [comment]
> @alexanderschr Thank you for the Pull Request 👍 . Would be nice to have a Unit test with a `PreviewImage` in https://github.com/sulu/SuluHeadlessBundle/blob/0.x/Tests/Unit/Content/Serializer/MediaSerializerTest.php. Let me know if you have time to add this else we can do it.

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@alexandersch Thank you! Good Work 👍

**Comment by niklasnatter** [comment]
Thanks a lot for the PR - really appreciating this!

**Pull request merged**